homeassistant/data_entry_flow.py

```diff
@@ -436,6 +440,25 @@ async def _async_handle_step(
                 error_if_core=False,
             )
 
+        if (
+            result["type"] == FlowResultType.SHOW_PROGRESS
+            and (progress_task := result.pop("progress_task", None))
+            and progress_task != flow.async_get_progress_task()
+        ):
+            # The flow's progress task was changed, register a callback on it
+            async def call_configure() -> None:
+                with suppress(UnknownFlow):
+                    await self.async_configure(flow.flow_id)
+
+            def schedule_configure(_: asyncio.Future) -> None:
```

Comment:
```suggestion
            def schedule_configure(_: asyncio.Task) -> None:
```

Reply:
I thought mypy would not like that, but it does indeed, thanks Martin 👍